File: PHP/LeftAndRightSumDifferences.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

class LeftAndRightSumDifferences {    

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function leftRightDifference($nums) {
        $l = 0;
        $r = array_sum($nums);
        $n = count($nums);
        $ans = array_fill(0, $n, 0);

        for ($i = 0; $i < $n; $i++) {
            $x = $nums[$i];
            $r -= $x;
            $ans[$i] = abs($l - $r);
            $l += $x;
        }
        return $ans;
    }

}

File: GO/LeftAndRightSumDifferences.go
/**
 * @version GO 1.26.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

func leftRightDifference(nums []int) []int {
	l, r := 0, 0
	for _, x := range nums {
		r += x
	}
	n := len(nums)
	ans := make([]int, n)
	for i, x := range nums {
		r -= x
		ans[i] = absLR(l - r)
		l += x
	}
	return ans
}

func absLR(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
